### ISSUE TITLE
OFFLOAD mode issue

### ISSUE THREAD

**Issue by jdorris1331** [hfp/libxsmm#28]
I am trying to run this on a Phi and am compiling with
    > make install OFFLOAD=1 MNK="2,4,6,8,10,12,14,16,18,20,23" AVX=3
but it errors out with

../../include/libxsmm.f90(143): error #6643: This statement is incorrectly positioned.
    !DIR$ ATTRIBUTES OFFLOAD:MIC :: libxsmm_smm_2_2_2
----------^
../../include/libxsmm.f90(150): error #6643: This statement is incorrectly positioned.
    !DIR$ ATTRIBUTES OFFLOAD:MIC :: libxsmm_dmm_2_2_2
----------^

Is there another flag that needs to be set to compile for the Phi?

**Comment by hfp** [comment]
Hello

I have reproduced the problem, and it is now fixed in the *Master* branch of LIBXSMM.
Thank you for reporting this issue!

One more comment, the build request:
> make install OFFLOAD=1 MNK="2,4,6,8,10,12,14,16,18,20,23" AVX=3

seems to aim for an AVX-512 host/main code path (KNL) along with an OFFLOAD code path (KNC). Is this really intended?

--
**Backgound**:
Using OFFLOAD=1 is always adding the offload code path (KNC) in order to complement whatever host code path is picked or requested (SSE, AVX) i.e., a hybrid library is built (`lib/intel64`). In addition, OFFLOAD=1 implies MIC=1 which also builds a KNC-only ("native") library for convenience (`lib/mic`). Right now, it appears the Intel Compiler does not support the combination of KNL+KNC within one hybrid binary. Anyways, I guess this was not your intention. Btw, the MIC/KNC instruction set is called IMCI (or KNCni) and *not* AVX-512 (perhaps that is the source of confusion?).
